docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -0,0 +1,492 @@
+
+* **Vault**: As Vault's database is encrypted, Vault cannot natively bootstrap its own TLS certificate.  Requires TLS certificate to be injected into container and its location set in a configuration file.
+
+* **PostgreSQL**: Requires TLS certificate to be injected into '$PGDATA' (default: `/var/lib/postgresql/data`) which is where the writable database files are kept.
+
+* **Kong (admin)**: Requires environment variable to be set to secure admin port with TLS, with a TLS certificates injected into the container.
+
+* **Kong (external)**: Requires a bring-your-own (BYO) external certificate,
+or as a fallback, a default one should be generated using a configurable external hostname.  (The Kong
+[ACME plugin](https://docs.konghq.com/hub/kong-inc/acme/)
+could possibly be used to automate this process.)
+
+* **Redis (v6)**: Requires TLS certificate set by configuration file or command line, with a TLS certificate injected into the container.
+
+* **Mosquitto**: Requires TLS certificate set by configuration file, with a TLS certificate injected into the container.
+
+Additionally, every EdgeX *microservice consumer* will require access to the root CA
+for certificate verification purposes,
+and every EdgeX *microservice server* will need a TLS leaf certificate and private key.
+
+There are no current plans for mutual auth TLS.
+
+#### Database passwords
+
+There are two databases currently used in EdgeX Hanoi:
+
+* **PostgreSQL** is the storage backend for Kong.
+  EdgeX provisions a PostgeSQL container with a superuser
+  account and password, and provides the same to Kong.
+  This password can be a derived secret or random secret.
+  Vault should be configured to rotate the
+  root password imediately after installation
+  (see this [example](https://github.com/Kong/docker-kong)),
+  with the password injected at runtime into Kong.
+
+* **Redis**: is the general-purpose database for EdgeX Hanoi.
+  Redis supports only a single password.
+  This password should be provisioned into the secret store
+  before services start running in production mode.
+  This password can be a derived secret or random secret.
+
+#### Kong admin credential
+
+The Kong admin port can be secured via
+[Kong API Loopback](https://docs.konghq.com/2.0.x/secure-admin-api/#introduction).
+After initial configuration of Kong,
+an authentication credential can be generated (derived or random)
+and stored in the secret store,
+and an authentication plugin such as the HTTP Basic authentication plugin
+can be applied to the route.
+
+### Summary of bootstrapping secrets
+
+In summary, the bootstrapping secrets are:
+
+* Secrets seed (supplied externally to the framework in some hardened configurations)
+* Consul Serf encryption key (used to control cluster membership)
+* Consul management token (used to configure Vault)
+* Consul ACL token (used to configure Vault if Consul is used as Vault's backing store)
```

Comment:
Per other comment, should these be removed if plan is to use `raft storage`?

Reply:
yes.